### ISSUE TITLE
[tune] AxSearch issues with max_concurrency

### ISSUE THREAD

**Issue by vakker** [ray-project/ray#8284]
### What is the problem?
Using `AxSearch` throws 
```
  File ".../ray/python/ray/tune/suggest/ax.py", line 83, in suggest
    if len(self._live_trial_mapping) >= self.max_concurrent:
AttributeError: 'AxSearch' object has no attribute '_live_trial_mapping'
```

Ray: latest master (76e1a4df)
Python: 3.7
OS: Ubuntu 18.04

### Reproduction (REQUIRED)
Just run the Ax example from [here](https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/ax_example.py).

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

I can see that there was a fix for other issues around refactoring (see commit 07daff8) when `max_concurrency` is passed in. If I run `AxSearch` with `max_concurrency=None`, then `AxSearch` throws an error because of [this](https://github.com/vakker/ray/blob/master/python/ray/tune/suggest/ax.py#L65). 

So maybe allow `max_concurrency=None` in `AxSearch`? Tbh it would be better to get rid of it completely, because it's very confusing when it is being used and when it's just there for legacy reasons.

If I remove `assert type(max_concurrent) is int and max_concurrent > 0` and just set it to `None` then the example works.

**Comment by richardliaw** [comment]
@vakker thanks a bunch for raising the issue! I just posted a fix https://github.com/ray-project/ray/pull/8285. 

max_concurrent for AxSearch should be an integer, and it is one of the only algorithms that leverage `max_concurrent` still (so it's not legacy yet 😄 ). Sorry for the confusion and again, thanks a bunch for taking the time to raise this issue.

**Comment by vakker** [comment]
@richardliaw great, thanks!
So how does `AxSearch` interact with `ConcurrencyLimiter`?
How is the following supposed to be?
```python
client = AxClient(enforce_sequential_optimization=False)
client.create_experiment(parameters=parameters)
search_alg = AxSearch(client, max_concurrent=?)
search_alg = ConcurrencyLimiter(search_alg, max_concurrent=?)
```

**Comment by richardliaw** [comment]
Hm, good point - I guess what we would want to do is:

```python
client = AxClient(enforce_sequential_optimization=False)
client.create_experiment(parameters=parameters)
search_alg = AxSearch(client)
search_alg = ConcurrencyLimiter(search_alg, max_concurrent=10)
```

**Comment by vakker** [comment]
Yeah, that's what I would assume.

What is the idea behind `ConcurrencyLimiter`? To reduce code duplication across different `Search` implementations or something else?

**Comment by richardliaw** [comment]
This is addressed in #8285. @vakker to me it serves two purposes - to reduce code duplication and also to isolate concerns away from the Searcher, so that the searcher code is easier to reason about.

**Comment by vakker** [comment]
@richardliaw yeah, that makes sense. The only thing that I can see is that `_live_trial_mapping` is used in many different ways in different Search algos, which might be a pain to factor out to a generic wrapper, but I don't know much about the planned design, so this might not be an issue at all.
